sympy/sets/sets.py

```diff
@@ -1962,15 +1962,23 @@ def _eval_powerset(self):
         return self.func(*[self.func(*s) for s in subsets(self.args)])
 
     def __ge__(self, other):
+        if not isinstance(other, Set):
+            raise TypeError("Invalid comparison of set with %s" % type(other))
```

Comment:
I used something like this in another PR -- what do you think about using this instead of bare `type(foo)`?

```python
def func(x):
    '''return function name of `x` (if defined) else the `type(x)`.'''
    return getattr(getattr(x, 'func', x), '__name__', type(x)))
```